staging/src/k8s.io/apiserver/pkg/server/options/serving_with_loopback.go

```diff
@@ -71,7 +71,19 @@ func (s *SecureServingOptionsWithLoopback) ApplyTo(secureServingInfo **server.Se
 
 	default:
 		*loopbackClientConfig = secureLoopbackClientConfig
-		(*secureServingInfo).SNICerts[server.LoopbackClientServerNameOverride] = &tlsCert
+
+		existingCallback := (*secureServingInfo).GetConfigForClient
+		(*secureServingInfo).GetConfigForClient = func(info *tls.ClientHelloInfo) (*tls.Config, error) {
+			config, err := existingCallback(info)
+			if err != nil {
+				return nil, err
+			}
+
+			config.NameToCertificate[server.LoopbackClientServerNameOverride] = &tlsCert
+			config.Certificates = append(config.Certificates, tlsCert)
```

Comment:
does this get called on every connection? is this modifying the cached *tls.Config that was returned for the previous connection?

Reply:
Sadly yes. It's tricky to maintain the existing situation where this function modifies the config, given we are moving to a callback approach. Some ideas:
1) Cache here as well
2) In the function where we build the tls.Config, iterate over some list of callbacks that is appended to in this function, which mutates the config

I am interested to hear your thoughts on how we could solve this